drivers/sensor/lsm6dso/lsm6dso.c

```diff
@@ -330,6 +355,46 @@ static int lsm6dso_sample_fetch_gyro(const struct device *dev)
 	return 0;
 }
 
+#if defined(CONFIG_LSM6DSO_ENABLE_FIFO)
+static int lsm6dso_sample_fetch_fifo(const struct device *dev)
```

Comment:
It seems plausible that in a k_work trigger handler sensor_fetch is called, blocking a work queue for a not insignificant amount of time given all the spi/i2c transactions that would occur here potentially.